examples/nlp/duplex_text_normalization/duplex_text_normalization_test.py

```diff
@@ -0,0 +1,118 @@
+        decoder_pretrained_model=PATH_TO_TRAINED_DECODER
+        data.test_ds.data_path=PATH_TO_TEST_FILE
+        mode={tn,itn,joint}
+
+The script also supports the `interactive` mode where a user can just make the model
+run on any input text:
+# python duplex_text_normalization_test.py
+        tagger_pretrained_model=PATH_TO_TRAINED_TAGGER
+        decoder_pretrained_model=PATH_TO_TRAINED_DECODER
+        mode={tn,itn,joint}
+        inference.interactive=true
+
+This script uses the `/examples/nlp/duplex_text_normalization/conf/duplex_tn_config.yaml`
+config file by default. The other option is to set another config file via command
+line arguments by `--config-name=CONFIG_FILE_PATH'.
+
+"""
+
+
+import numpy as np
+import nemo.collections.nlp.data.text_normalization.constants as constants
+
+from tqdm import tqdm
+from math import ceil
+from nltk import word_tokenize
+from time import perf_counter
+from omegaconf import DictConfig, OmegaConf
+from utils import TAGGER_MODEL, DECODER_MODEL, instantiate_model_and_trainer
+
+from nemo.utils import logging
+from nemo.core.config import hydra_runner
+from nemo.collections.nlp.models import DuplexTextNormalizationModel
+from nemo.collections.nlp.data.text_normalization import TextNormalizationTestDataset
+
+@hydra_runner(config_path="conf", config_name="duplex_tn_config")
+def main(cfg: DictConfig) -> None:
+    logging.info(f'Config Params: {OmegaConf.to_yaml(cfg)}')
+    tagger_trainer, tagger_model = instantiate_model_and_trainer(cfg, TAGGER_MODEL, False)
+    decoder_trainer, decoder_model = instantiate_model_and_trainer(cfg, DECODER_MODEL, False)
+    tn_model = DuplexTextNormalizationModel(tagger_model, decoder_model)
+
+    if not cfg.inference.interactive:
+        # Setup test_dataset
+        test = TextNormalizationTestDataset(cfg.data.test_ds.data_path,
+                                            cfg.data.test_ds.mode)
+
+        # Apply the model on the test dataset
+        all_dirs, all_inputs, all_preds, all_targets, all_run_times = [], [], [], [], []
+        batch_size = cfg.data.test_ds.batch_size
+        nb_iters = int(ceil(len(test) / batch_size))
+        for i in tqdm(range(nb_iters)):
+            start_idx = i * batch_size
+            end_idx = (i+1) * batch_size
+            batch_insts = test[start_idx:end_idx]
+            batch_dirs, batch_inputs, batch_targets = zip(*batch_insts)
+            # Inference and Running Time Measurement
+            batch_start_time = perf_counter()
+            batch_preds = tn_model._infer(batch_inputs, batch_dirs)
+            batch_run_time = (perf_counter() - batch_start_time) * 1000  # milliseconds
+            all_run_times.append(batch_run_time)
+            # Update all_dirs, all_inputs, all_preds and all_targets
+            all_dirs.extend(batch_dirs)
+            all_inputs.extend(batch_inputs)
+            all_preds.extend(batch_preds)
+            all_targets.extend(batch_targets)
```

Comment:
Can this be moved inside the model as a separate function?
Could you save errors after the eval is done to make error analysis easy?

Reply:
Yes. I moved that inside the model and also added code to save the errors.